src/dodal/beamlines/i22.py

```diff
@@ -41,338 +39,233 @@
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def slits_2():
+    """Create Slits instance for slot 2."""
+    return numbered_slits(2)
 
 
-def slits_3(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> Slits:
-    return numbered_slits(
-        3,
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def slits_3():
+    """Create Slits instance for slot 3."""
+    return numbered_slits(3)
 
 
-def slits_4(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> Slits:
-    return numbered_slits(
-        4,
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@skip_device()
+def slits_4():
+    """Create Slits instance for slot 4."""
+    return numbered_slits(4)
 
 
-def slits_5(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> Slits:
-    return numbered_slits(
-        5,
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def slits_5():
+    """Create Slits instance for slot 5."""
+    return numbered_slits(5)
 
 
-def slits_6(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> Slits:
-    return numbered_slits(
-        6,
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def slits_6():
+    """Create Slits instance for slot 6."""
+    return numbered_slits(6)
 
 
-def fswitch(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> FSwitch:
-    return device_instantiation(
-        FSwitch,
-        "fswitch",
-        "-MO-FSWT-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def fswitch():
+    """Create an FSwitch instance with specific settings."""
+    return FSwitch(
+        name="fswitch",
+        prefix="-MO-FSWT-01:",
         lens_geometry="paraboloid",
         cylindrical=True,
         lens_material="Beryllium",
     )
 
 
-# Must document what PandAs are physically connected to
-# See: https://github.com/bluesky/ophyd-async/issues/284
-def panda1(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> HDFPanda:
-    return device_instantiation(
-        HDFPanda,
-        "panda1",
-        "-EA-PANDA-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def panda1():
+    """Create HDFPanda instance for panda1."""
+    return HDFPanda(
+        name="panda1",
+        prefix="-EA-PANDA-01:",
         directory_provider=get_directory_provider(),
     )
 
 
 @skip_device()
```

Comment:
add `skip: bool = False` to `@device_factory`?

Reply:
yes, will